src/core/ddsi/src/q_entity.c

```diff
@@ -5518,7 +5515,7 @@ int new_proxy_writer (struct ddsi_domaingv *gv, const struct ddsi_guid *ppguid,
 
 #ifdef DDSI_INCLUDE_SECURITY
   set_proxy_writer_security_info(pwr, plist);
-  q_omg_get_proxy_writer_security_info(pwr, plist, &(pwr->security_info));
+  q_omg_get_proxy_writer_security_info(pwr, plist, &(pwr->c.security_info));
```

Comment:
I'd rather see this call moved to ``proxy_endpoint_common_init`` — and then using ``q_omg_get_proxy_endpoint_security_info`` which is how ``q_omg_get_proxy_writer_security_info`` and ``q_omg_get_proxy_reader_security_info`` are implemented anyway.